AK/Error.h

```diff
@@ -19,12 +21,27 @@
 
 namespace AK {
 
+template<typename T, typename E>
+class [[nodiscard]] ErrorOr;
+
 class Error {
 public:
+    Error(Error&&);
+    Error(Error&);
+    Error(Error const&);
+
+    ~Error();
```

Comment:
Have you checked the performance impact of this? Error not being a POD type will cause all sorts of optimisations to go away.